contracts/halo-pool/src/state.rs

```diff
@@ -127,42 +88,26 @@ pub enum TokenInfoRaw {
     NativeToken { denom: String },
 }
 
-impl TokenInfo {
-    pub fn is_token(&self) -> bool {
-        matches!(self, TokenInfo::Token { .. })
-    }
-
-    pub fn is_native_token(&self) -> bool {
-        matches!(self, TokenInfo::NativeToken { .. })
-    }
-
-    pub fn to_raw(&self, api: &dyn Api) -> StdResult<TokenInfoRaw> {
-        match self {
-            TokenInfo::NativeToken { denom } => Ok(TokenInfoRaw::NativeToken {
-                denom: denom.to_string(),
-            }),
-            TokenInfo::Token { contract_addr } => Ok(TokenInfoRaw::Token {
-                contract_addr: api.addr_validate(contract_addr)?,
-            }),
-        }
-    }
-}
-
 // We define a custom struct for each query response
 #[cw_serde]
 pub struct PoolInfo {
-    pub staked_token: String,
-    pub reward_token: TokenInfo,
     pub reward_per_second: Decimal,
     pub start_time: u64,
     pub end_time: u64,
     pub pool_limit_per_user: Option<Uint128>,
-    pub whitelist: Vec<Addr>,
+    // Total staked token at end time to calculate the reward amount correctly
+    //if the pool has multiple phases.
+    pub total_staked_at_end_time: Uint128,
 }
 
-// We define a custom struct for each query response
 #[cw_serde]
-pub struct PoolResponse {
+pub struct PoolInfos {
     pub staked_token: String,
-    pub total_share: Uint128,
+    pub reward_token: TokenInfo,
+    pub current_phase_index: u64,
+    pub pool_infos: Vec<PoolInfo>,
```

Comment:
pool_infos, whitelist, reward_balance, last_reward_time, accrued_token_per_share gom lại thành 1 struct rồi để chỗ này thành 1 vector của 1 struct có được ko

Reply:
Moved whitelist, reward_balance, last_reward_time and accrued_token_per_share from PoolInfos to PhaseInfo struct